src/main/java/io/nats/client/ConsumerContext.java

```diff
@@ -16,18 +16,135 @@
 import io.nats.client.api.ConsumerInfo;
 
 import java.io.IOException;
+import java.time.Duration;
 
 /**
+ * The Consumer Context provides a convenient interface around a defined JetStream Consumer
  * SIMPLIFICATION IS EXPERIMENTAL AND SUBJECT TO CHANGE
  */
 public interface ConsumerContext {
-    String getName();
+    /**
+     * Gets the consumer name that was used to create the context.
+     * @return the consumer name
+     */
+    String getConsumerName();
+
+    /**
+     * Gets information about the consumer behind this subscription.
+     * @return consumer information
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
     ConsumerInfo getConsumerInfo() throws IOException, JetStreamApiException;
```

Comment:
Do you think it would be valuable to provide e.g. `getCachedConsumerInfo()` to be able to get to the consumer info returned when executing `consumerContext()` or `addConsumer()` without an extra call to the server?

Reply:
I added a getCachedConsumerInfo that returns the last one that was read from the server. I added to the doc that technically it could be out of date.